ydb/core/tx/schemeshard/schemeshard_import.cpp

```diff
@@ -36,6 +36,124 @@ namespace {
+
+        Y_ABORT_UNLESS(itemIdx < importInfo.Items.size());
+        const auto& item = importInfo.Items.at(itemIdx);
+
+        const auto opId = TOperationId(item.WaitTxId, FirstSubTxId);
+        if (item.WaitTxId != InvalidTxId && ss->TxInFlight.contains(opId) &&
+            ss->TxInFlight.at(opId).TxType == TTxState::TxRestore) {
+            const auto& txState = ss->TxInFlight.at(opId);
+
+            itemProgress.set_parts_total(itemProgress.parts_total() + txState.Shards.size());
+            itemProgress.set_parts_completed(itemProgress.parts_completed() + txState.Shards.size() - txState.ShardsInProgress.size());
+            *itemProgress.mutable_start_time() = SecondsToProtoTimeStamp(txState.StartTime.Seconds());
+        } else {
+            if (!ss->Tables.contains(item.DstPathId)) {
+                return;
+            }
+
+            auto table = ss->Tables.at(item.DstPathId);
+            auto it = table->RestoreHistory.end();
+            if (item.WaitTxId != InvalidTxId && table->RestoreHistory.contains(item.WaitTxId)) {
+                it = table->RestoreHistory.find(item.WaitTxId);
+            } else if (table->RestoreHistory.size() == 1) {
+                // As restore operations always create new table, case table->RestoreHistory.size() > 1 is unexpected
+                // To return at least something in this case, we'll just pick one of items from RestoreHistory
+                it = table->RestoreHistory.begin();
+            }
+
+            if (it == table->RestoreHistory.end()) {
+                return;
+            }
+
+            const auto& restoreResult = it->second;
+            itemProgress.set_parts_total(itemProgress.parts_total() + restoreResult.TotalShardCount);
+            itemProgress.set_parts_completed(itemProgress.parts_completed() + restoreResult.TotalShardCount);
+            *itemProgress.mutable_start_time() = SecondsToProtoTimeStamp(restoreResult.StartDateTime);
+            // Update for case no index builds are required
+            *itemProgress.mutable_end_time() = SecondsToProtoTimeStamp(restoreResult.CompletionDateTime);
+        }
+    }
+
+    void AddBuildIndexesItemProgress(TSchemeShard* ss, const TImportInfo& importInfo, ui32 itemIdx, 
+        i32 indexIdx, Ydb::Import::ImportItemProgress& itemProgress) {
+
+        Y_ABORT_UNLESS(itemIdx < importInfo.Items.size());
+        const auto& item = importInfo.Items.at(itemIdx);
+
+        if (!item.Table) {
+            return;
+        }
+        Y_ABORT_UNLESS(indexIdx < item.Table->indexes_size());
+
+        const ui32 partsTotal = ss->Tables.contains(item.DstPathId) ?
+            ss->Tables.at(item.DstPathId)->GetPartitions().size() :
+            GetTablePartsFromRequest(*item.Table);
+
+        const auto buildUid = MakeIndexBuildUid(importInfo, itemIdx, indexIdx);
+        if (ss->IndexBuildsByUid.contains(buildUid)) {
+            const auto& indexBuild = ss->IndexBuildsByUid.FindPtr(buildUid)->Get();
```

Comment:
Проще так сделать, так как ты строчку назад проверил наличие

```suggestion
            const auto& indexBuild = ss->IndexBuildsByUid[buildUid];
```